packages/waku/src/lib/vite-plugins/fs-router-typegen.ts

```diff
@@ -73,18 +73,52 @@ export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
       outputFile = joinPath(config.root, opts.srcDir, 'pages.gen.ts');
     },
     configureServer(server) {
-      if (
-        !entriesFilePossibilities ||
-        !pagesDir ||
-        !outputFile ||
-        entriesFilePossibilities.some((entriesFile) =>
-          existsSync(entriesFile),
-        ) ||
-        !existsSync(pagesDir)
-      ) {
+      if (!entriesFilePossibilities || !pagesDir || !outputFile) {
         return;
       }
 
+      if (!existsSync(pagesDir)) {
+        return;
+      }
+
+      const existingServerEntry = entriesFilePossibilities.find((entriesFile) =>
+        existsSync(entriesFile),
+      );
+
+      if (existingServerEntry) {
+        try {
+          const file = swc.parseSync(
+            readFileSync(existingServerEntry, 'utf8'),
+            {
+              syntax: 'typescript',
+              tsx: true,
+            },
+          );
+
+          const usesCreatePages = file.body.some((node) => {
+            if (node.type === 'ImportDeclaration') {
+              if (node.source.value !== 'waku') {
```

Comment:
It's technically possible and valid to do `import { fsRouter } from 'waku/router/server'`.

Reply:
That should still generate types I think because it's still using filesystem routing. 

With this logic that would still run the generation because we only check for createPages being used to disable generation.

Reply:
Oh, this is `usesCreatePages`. Got it.

Then, my concern is that we shouldn't run generation for this case: https://github.com/wakujs/waku/blob/fe8fb224abbaba0fa962782a25690264945e1114/examples/31_minimal/src/server-entry.tsx

And, also `import { createPages } from 'waku/router/server'` should be detected.

Reply:
thanks! updated to match that logic